### PR TITLE
Adding helpers used in enrichment-processor

### PR DESCRIPTION
**Pull request by robert-pierce**
[Card](https://rentpath.atlassian.net/browse/SRV-2322)

### CODE REVIEW

File: src/rp/jackdaw/processor_supervisor.clj [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   :thread-names (thread-names processor)
+   :topic-registry topic-registry})
+
+(defn- maybe-handle-thread-exception
+  [{:keys [state-atom] :as this} ex-thread throwable maybe-update-schema]
+  (let [{:keys [processor thread-names topic-registry]} @state-atom
+        ex-thread-name (.getName ex-thread)]
+    (when (contains? thread-names ex-thread-name)
+      (log/info "Attempting to handle thread exception..." {:thread-name ex-thread-name})
+      (when-let [new-topic-registry (maybe-update-schema topic-registry (:brand processor))]
```
Comment (sbrauer) [inline src/rp/jackdaw/processor_supervisor.clj:29]: Assumes the existence of a `:brand`.  
I wonder if this (and perhaps also having to pass `maybe-update-schema` to `handle-thread-exception`) are indicators that this isn't quite general enough.

Comment (robert-pierce) [reply src/rp/jackdaw/processor_supervisor.clj:29]: Good catch on the `brand`.

Comment (robert-pierce) [reply src/rp/jackdaw/processor_supervisor.clj:29]: Let me see if I can tinker with it to push the maybe-update-schema to the processor apps

Comment (awebneck) [reply src/rp/jackdaw/processor_supervisor.clj:29]: Maybe designate a maybe-update-fn or something similar via config?

**Review by sbrauer — APPROVED** [review]
LGTM

**Pull request merged**